signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
+					return PolicyRequirementError(fmt.Sprintf("Signature for identity %s is not accepted", ref))
+				}
+				return nil
```

Comment:
alright, clear, so, when the user requests an image by digest this callback is called and `matchesDockerReference` is called with that `sb` string and do an `if/else` based on whether the image requested is canonical or not (I don't think it can happen to receive a request for `reference.IsNameOnly` right?)